modules/020-deckhouse/hooks/update_deckhouse_image.go

```diff
@@ -404,14 +414,41 @@ func (du *deckhouseUpdater) releaseChannelUpdate(input *go_hook.HookInput) error
 	return nil
 }
 
-func applyRelease(input *go_hook.HookInput, rl deckhouseRelease, ts time.Time) {
+func checkReleaseRequirements(input *go_hook.HookInput, rl deckhouseRelease, ts time.Time) bool {
+	for key, value := range rl.Requirements {
+		passed, err := requirements.CheckRequirement(key, value, input.Values)
+		if !passed {
+			msg := fmt.Sprintf("%q requirement for DeckhouseRelease %q not met: %s", key, rl.Version, err)
+			st := statusPatch{
+				Phase:          v1alpha1.PhasePending,
+				Message:        msg,
+				Approved:       true,
+				TransitionTime: ts,
+			}
+			input.PatchCollector.MergePatch(st, "deckhouse.io/v1alpha1", "DeckhouseRelease", "", rl.Name, object_patch.WithSubresource("/status"))
+			return false
+		}
+	}
+
+	return true
+}
+
+func applyRelease(input *go_hook.HookInput, rl deckhouseRelease, ts time.Time) (applied bool) {
+	// check release dependencies
+	passed := checkReleaseRequirements(input, rl, ts)
+	if !passed {
+		input.MetricsCollector.Set("d8_release_blocked", 1, map[string]string{"name": rl.Name}, metrics.WithGroup("d8_releases"))
```

Comment:
Do we need to clear metric if release will be accepted?

Reply:
In the beginning of this hook we have group cleanup
```
input.MetricsCollector.Expire("d8_releases")
```

Reply:
ok. I saw it